src/frontend/picker.ts

```diff
@@ -0,0 +1,158 @@
+
+    getItems(){
+        return this.getEntries().then(function (processEntries: any[]) {
+            return processEntries.map(function (p) { return p.toAttachItem(); });
+        });
+    };
+
+    pickItem(launchConfig){
+        this.launchConfig = launchConfig;
+
+        var attachPickOptions = {
+            matchOnDescription: true,
+            matchOnDetail: true,
+            placeHolder: "Select the docker name to attach to"
+        };
+        return vscode.window.showQuickPick(this.getItems(), attachPickOptions)
+            .then(function (dockerItem: any) {
+                return dockerItem ? dockerItem.id : null;
+            });
+    };
+
+    parseItems(items: string){
+    //Parses a line of the output and returns array of objects
+        var lines = items.split('\n');
+        var entries = [];
+        for (var i = 1; i < lines.length; i++) {
+            var line = lines[i];
+            if (!line) {
+                continue;
+            }
+            var docker_1 = this.lineParser(line);
+            entries.push(docker_1);
+        }
+        return entries;
+    }
+
+    execChildProcess(process, workingDirectory) {
+        return new Promise(function (resolve, reject) {
+            child_process.exec(process, { cwd: workingDirectory, maxBuffer: 512000 }, function (error, stdout, stderr) {
+                if (error) {
+                    reject(error);
+                    return;
+                }
+                if (stderr && stderr.length > 0) {
+                    reject(new Error(stderr));
+                    return;
+                }
+                resolve(stdout);
+            });
+        });
+    };
+
+    abstract getEntries(): any; //any[]
+    abstract lineParser(line: string): any;
+}
+
+export class DockerPicker extends Picker {
+    getEntries() {
+        var psCommand = ("docker ps --format {{.ID}}@{{.Names}}@{{.Image}}@{{.Labels}}");
+        var _this = this
+        return this.execChildProcess(psCommand, null).then(function (items: string) {
+            return _this.parseItems('\n'+items); //Add bogus header line
+        });
+    }
+
+    lineParser(line: string) {
+        var dockerEntry = line.split('@');
+        dockerEntry.push(dockerEntry.splice(3).join('@'));
+        //Combine all entries after 4 in case labels can have @
+        dockerEntry = dockerEntry.map(function(v){return v.trim();});
+        return new DockerItem(dockerEntry[0], dockerEntry[1], dockerEntry[2], dockerEntry[3]);
+    }
+}
+
+export class DockerPidPicker extends Picker {
+    static secondColumnCharacters: number = 50;
+
+    getEntries() {
+        var _this = this;
+
+        if (this.launchConfig.dockerName === "null"){return;}
+
+        var dockerName = this.launchConfig.dockerName;
+
+        var commColumnTitle = Array(DockerPidPicker.secondColumnCharacters).join("a");
+        var psCommand = ("docker exec "+dockerName+" ps -axww -o pid=,comm=" + commColumnTitle + ",args=");
```

Comment:
Can dockerName or commColumnTitle contain spaces? If yes then you should probably add quotes around it or use the overload that takes an array of values. I prefer the latter more because you have total control over spawning the process and don't need to worry about spaces or having some weird shell that doesn't support this syntax for some reason